filebeat/crawler/registrar.go

```diff
@@ -91,7 +94,7 @@ func (r *Registrar) Run() {
 			return
 		// Treats new log files to persist with higher priority then new events
 		case state := <-r.Persist:
-			r.State[*state.Source] = state
+			r.setState(*state.Source, state)
```

Comment:
That is older code. I think it does not matter, but would prefer to change it in a later PR to have as few changes as possible.